workflow/automation/lib/MgmtDB.py

```diff
@@ -53,7 +53,11 @@ def connect_db_ctx(db_file, verbose=False):
 
     A commit is run at the end of the context.
     """
-    conn = sql.connect(db_file)
+    # timeout parameter specifies how long the connection should wait for the lock to
+    # go away until raising an exception. Default is 5 secs
+    # https://stackoverflow.com/a/8618328/2005856
+    # https://docs.python.org/2/library/sqlite3.html#sqlite3.connect
+    conn = sql.connect(db_file, timeout=10)
```

Comment:
I don't like the `import sqlite3 as sql`. I'd rather see `sqlite3.connect` and at least know it came from the sqlite3 module. When I saw this I thought another library might be involved.

Reply:
Also, what happens here if the connection is opened but then the rest of this function fails? In this case a connection to the database (and potentially a lock) would be leaked. Don't know if it affects your DB lock problems, but maybe in the interest of cleaning up all loose ends this connection should be freed in the case of an error with the rest of the function.

Reply:
Oh ok nevermind I can't read that is accounted for.

Reply:
ok @sungeunbae if you don't want to change that because of pegasus that's fine. Maybe link to the latest sqlite3 docs rather than Python 2 docs? https://docs.python.org/3/library/sqlite3.html#sqlite3.connect

Reply:
Good suggestion!